transport/peer/x/peerlist/roundrobin/roundrobin.go

```diff
@@ -47,7 +48,8 @@ func New(peerIDs []transport.PeerIdentifier, agent transport.Agent) (*RoundRobin
 type RoundRobin struct {
 	lock sync.Mutex
 
-	pr                 *PeerRing
+	nonAvailablePeers  map[string]transport.Peer
```

Comment:
hmm, i didn't want to do that because there was a weird disconnect where "unavailable" and "connecting" were sort-of separate, so I didn't want to use the wording "unavailable"